frontend/src/pages/projects/screens/spawner/useImageStreams.ts

```diff
@@ -4,14 +4,15 @@ import { getNotebookImageStreams } from '~/api';
 
 const useImageStreams = (
   namespace?: string,
+  includeDisabled?: boolean,
 ): [imageStreams: ImageStreamKind[], loaded: boolean, loadError: Error | undefined] => {
   const [imageStreams, setImageStreams] = React.useState<ImageStreamKind[]>([]);
   const [loaded, setLoaded] = React.useState(false);
   const [loadError, setLoadError] = React.useState<Error | undefined>(undefined);
 
   React.useEffect(() => {
     if (namespace) {
-      getNotebookImageStreams(namespace)
+      getNotebookImageStreams(namespace, includeDisabled)
```

Comment:
If an error occurs when `includeDisabled` changes and we have previously loaded data, `setImageStreams` will remain the old value as the `catch` clause does not reset the data. Also I don't think we should be setting `loaded` to `true` if we have an error. The typical pattern here sets `loaded` to `true` when data is available. Loaded state should be unaffected by error state.

Reply:
I can however set the state back to an empty array to stop previous loaded data from lingering on error

Reply:
actually i am just going to switch this over to useFetchState if i can so it uses that common logic